src/odemis/acq/test/stream_test.py

```diff
@@ -2668,6 +2668,98 @@ def test_cl_ss(self):
     def _on_image(self, im):
         self._image = im
 
+# @skip("faster")
+class SettingsStreamsTestCasePolAnalyzer(unittest.TestCase):
+    """
+    Tests of the *SettingsStreams, to be run with a (simulated) SPARC including polarization
+    analyzer hardware. Testing for live view.
+    """
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        # try:
+        #     test.start_backend(SPARC_CONFIG)
+        # except LookupError:
+        #     logging.info("A running backend is already found, skipping tests")
+        #     cls.backend_was_running = True
+        #     return
+        # except IOError as exp:
+        #     logging.error(str(exp))
+        #     raise
+
+        # Find CCD & SEM components
+        cls.ccd = model.getComponent(role="ccd")
+        cls.spec = model.getComponent(role="spectrometer")
+        cls.spgp = model.getComponent(role="spectrograph")
+        cls.ebeam = model.getComponent(role="e-beam")
+        cls.sed = model.getComponent(role="se-detector")
+        cls.analyzer = model.getComponent(role="pol-analyzer")
+
+        mic = model.getMicroscope()
+        cls.optmngr = path.OpticalPathManager(mic)
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        # test.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+    def test_arpol_ss(self):
+        """ Test ARSettingsStream """
+        # Create the stream
+        ars = stream.ARSettingsStream("test",
+                      self.ccd, self.ccd.data, self.ebeam, self.analyzer,
+                      detvas={"exposureTime", "readoutRate", "binning", "resolution"})
+        self._image = None
+        ars.image.subscribe(self._on_image)
+
+        # shouldn't affect
+        ars.roi.value = (0.15, 0.6, 0.8, 0.8)
+        ars.repetition.value = (5, 6)
+        ars.detExposureTime.value = 0.3  # s
+
+        # set analyzer to position different from polarization VA connected to GUI
+        ars.polarization._set_value("lhc")
+        self.analyzer.moveAbs({"pol": "rhc"})
+        # while stream is not active, HW should not move, therefore
+        # check polarization VA connected to GUI did not trigger position VA listening to HW
+        self.assertNotEqual(ars.polarization.value, self.analyzer.position.value.values()[0])
```

Comment:
For position, it's clearer and more reliable to write `self.analyzer.position.value["pol"]`.